python/alibiexplainer/alibiexplainer/explainer.py

```diff
@@ -71,9 +86,50 @@ def _predict_fn(self, arr: Union[np.ndarray, List]) -> np.ndarray:
                 instances.append(req_data.tolist())
             else:
                 instances.append(req_data)
+        shape = np.shape(instances)
         loop = asyncio.get_running_loop()  # type: ignore
-        resp = loop.run_until_complete(self.predict({"instances": instances}))
-        return np.array(resp["predictions"])
+
+        # Prepare the request beased on the predictor protocol
+        if self.protocol == PredictorProtocol.GRPC_V2.value:
```

Comment:
Can we add an e2e test ?

Reply:
e2e is possible but it requires ModelMesh installation and an issue to be addressed, https://github.com/kserve/modelmesh-serving/issues/254

Reply:
Please note the Alibi explainer output values might/will be different between requests due to some randomness in it's logic. e2e would prove functionally it works. Unfortunately I am not the Alibi expert to interpret the results though.

Reply:
ModelMesh has never been part of KServe e2e. It would take some thoughts, alignment, and work to get it right.  So for this PR I added a notebook for the Alibi cifar10 example with instructions for end to end execution of running the Alibi explainer with predictor hosted in ModelMesh.